lib/repeatable/schedule.rb

```diff
@@ -14,6 +14,8 @@ def initialize(arg)
     end
 
     def occurrences(start_date, end_date)
+      fail ArgumentError, 'end_date must be after start_date' if end_date < start_date
```

Comment:
This will fit just a little better after the `Date()` conversion methods below to ensure we're comparing `Date` objects.

With the check this spot, we'd end up with some false failures in a few edge cases.
```ruby
start_date = '20th Jan 2015'
end_date = '10th Feb 2015'

start_date < end_date              # => false
Date(start_date) < Date(end_date)  # => true
```